app/javascript/components/Home/Service/WorthPhotoContainer.1.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { Component } from "react";
+import { FadeInLeft } from "../../Constants/Constants";
+import {
+  postFetch,
+  deleteFetch,
+  getFetch
+} from "../../Constants/FetchComponent";
+import { DeleteButton, AddButton } from "../../Constants/Buttons";
+
+class WorthPhotoContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      url: "worth_photos",
+      photoData: [],
+      photo: "",
+      refreshKey: false
+    };
+  }
+
+  toggleRefreshKey = () => this.setState({ refreshKey: true });
+
+  onChange = e => this.setState({ [e.target.name]: e.target.value });
+
+  onSubmit = event => {
+    event.preventDefault();
+    const url = `/api/v1/${this.state.url}`;
+    const token = document.querySelector('meta[name="csrf-token"]').content;
+    const { photo } = this.state;
+    const body = { photo };
+
+    postFetch(url, token, body)
+      .then(this.toggleRefreshKey)
+      .catch(error => console.log("error message =>", error.message));
+  };
+
+  deleteEvent = id => {
+    const url = `/api/v1/${this.state.url}/${id}`;
+    const token = document.querySelector('meta[name="csrf-token"]').content;
+
+    deleteFetch(url, token)
+      .then(this.toggleRefreshKey)
+      .catch(error => console.log("error message =>", error.message));
+  };
+
+  componentDidMount() {
+    getFetch(this.state.url)
+      .then(body => this.setState({ photoData: body }))
+      .catch(error => console.log("error message =>", error.message));
+  }
+
+  componentDidUpdate() {
+    this.state.refreshKey
+      ? getFetch(this.state.url)
+          .then(body => this.setState({ photoData: body }))
+          .then(this.setState({ refreshKey: false }))
+          .catch(error => console.log("error message =>", error.message))
+      : null;
+  }
+
+  render() {
+    let photos = this.state.photoData.map(element => {
```

Comment:
You could make a PhotoComponent instead of having this here. Inside the photo component you could also have that handleDelete function, rather than having it defined here

Reply:
I completely agree, moving rendered data only to stateless.